### ISSUE TITLE
question about the inpaint_ops.py

### ISSUE THREAD

**Issue by kikyou123** [JiahuiYu/generative_inpainting#22]
In my opinion, the kernel size when extract patches to generate deconvolution filters (raw_w) should be the same with the kernel size to generate the convolution filters (w).  But why raw_w use `kernel=2*rate`, but w use `ksize`?

**Comment by kikyou123** [comment]
yes, but the kernel size means the patch size for a match, shouldn't they be same?

**Comment by JiahuiYu** [comment]
When the patch size is larger, the overlapped area is averaged. Basically larger patch mean more accurate of original resolution. You can actually try different sizes of code to compare the difference.

**Comment by kikyou123** [comment]
I am sorry, but i can not understand the 

>  larger patch mean more accurate of original resolution``

**Comment by kikyou123** [comment]
And I have another question, why `yi = tf.nn.conv2d_transpose(yi, wi_center, tf.concat([[1], raw_fs[1:]], axis=0), strides=[1,rate,rate,1]) / 4.`  divide 4?

**Comment by JiahuiYu** [comment]
Because the overlapped area is averaged (2x2=4), as mentioned above.. You'd better have a try to see difference.

**Comment by kikyou123** [comment]
I think I got it! Thank you for your patient explanation.

**Comment by 12ycli** [comment]
@kikyou123 Hello! Can you explain this for me ? I am struggling with this.

**Comment by 12ycli** [comment]
https://github.com/JiahuiYu/generative_inpainting/issues/294#issue-468741974
